griptape/tasks/base_task.py

```diff
@@ -29,26 +29,32 @@ class State(Enum):
     parent_ids: list[str] = field(factory=list, kw_only=True)
     child_ids: list[str] = field(factory=list, kw_only=True)
     max_meta_memory_entries: Optional[int] = field(default=20, kw_only=True)
+    structure: Optional[Structure] = field(default=None, kw_only=True)
 
     output: Optional[BaseArtifact] = field(default=None, init=False)
-    structure: Optional[Structure] = field(default=None, init=False)
     context: dict[str, Any] = field(factory=dict, kw_only=True)
     futures_executor_fn: Callable[[], futures.Executor] = field(
         default=Factory(lambda: lambda: futures.ThreadPoolExecutor()),
         kw_only=True,
     )
 
+    def __attrs_post_init__(self) -> None:
+        if self.structure is not None and self not in self.structure.tasks:
+            self.structure.add_task(self)
```

Comment:
Could you perform the `self not in self.structure.tasks` check inside `Structure.add_task()` instead of in the client code? (Enforcing the invariant of no duplicate tasks in the `tasks` list is really the responsibility of `Structure`)